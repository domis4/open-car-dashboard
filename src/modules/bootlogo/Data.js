import { connect } from 'react-redux'

const mapStateToProps = ({ router }) => ({
  pathname: router.location.pathname,
  hash: router.location.hash,
})

export default connect(mapStateToProps)
