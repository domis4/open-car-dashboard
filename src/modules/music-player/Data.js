import { compose, withState } from 'recompose'

export default compose(withState('tabValue', 'setTabValue', 'lists'))
