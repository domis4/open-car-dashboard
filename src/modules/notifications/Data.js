import { connect } from 'react-redux'
import { removeNotification } from '../../redux/notifications'

const mapStateToProps = ({ notifications }) => ({
  notifications,
})

const mapDispatchToProps = {
  removeNotification,
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)
