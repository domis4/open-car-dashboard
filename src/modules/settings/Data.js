import { compose, withState, lifecycle } from 'recompose'

export default compose(
  withState('tabValue', 'setTabValue', 'lists'),
  lifecycle({
    componentDidMount() {
      console.log('mounted')
    },
  })
)
