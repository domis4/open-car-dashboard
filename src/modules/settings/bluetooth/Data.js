import { withStateHandlers } from 'recompose'

export default withStateHandlers(
  () => ({ isBluetoothVisible: false }),
  ({ isScanning }) => ({
    handleBluetoothScanToggle: ({ isBluetoothVisible }) => () => {},
  })
)
