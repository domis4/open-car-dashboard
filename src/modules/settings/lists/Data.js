import { compose } from 'recompose'
import { connect } from 'react-redux'
import { push } from 'connected-react-router'

const mapStateToProps = ({ router }) => ({
  pathname: router.location.pathname,
  hash: router.location.hash,
})

const mapDispatchToProps = {
  callRoute: route => push(route),
}

export default compose(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )
)
