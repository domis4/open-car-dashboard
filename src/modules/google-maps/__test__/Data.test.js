import React from 'react'
import { compose } from 'recompose'
import { shallow } from 'enzyme'
import Data from '../Data'

describe('google-maps module Data.js', () => {
  it('expects component to match snapshot', () => {
    const Dummy = () => <div />
    const Component = compose(Data)(Dummy)
    const wrapper = shallow(<Component />)

    expect(wrapper).toMatchSnapshot()
  })

  it('expects component to receive a `setTabValue` function as prop', () => {
    const Dummy = () => <div />
    const Component = compose(Data)(Dummy)
    const wrapper = shallow(<Component />)

    expect(wrapper.props().setTabValue).toBeDefined()
  })
  it('expects component to receive a `tabValue` function as prop', () => {
    const Dummy = () => <div />
    const Component = compose(Data)(Dummy)
    const wrapper = shallow(<Component />)

    expect(wrapper.props().tabValue).toBeDefined()
    expect(wrapper.props().tabValue).toEqual('lists')
  })
})
