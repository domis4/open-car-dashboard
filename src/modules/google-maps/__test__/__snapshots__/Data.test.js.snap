// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`google-maps module Data.js expects component to match snapshot 1`] = `
<Dummy
  setTabValue={[Function]}
  tabValue="lists"
/>
`;
